src/Pim/Component/Catalog/Updater/Remover/MultiSelectAttributeRemover.php

```diff
@@ -53,46 +52,38 @@ public function removeAttributeData(
         $this->checkLocaleAndScope($attribute, $options['locale'], $options['scope'], 'multi-select');
         $this->checkData($attribute, $data);
 
-        $attributeOptions = [];
-        foreach ($data as $optionCode) {
-            $option = $this->getOption($attribute, $optionCode);
-            if (null === $option) {
-                throw InvalidArgumentException::arrayInvalidKey(
-                    $attribute->getCode(),
-                    'code',
-                    'The option does not exist',
-                    'remover',
-                    'multi select',
-                    $optionCode
-                );
-            }
-
-            $attributeOptions[] = $option;
-        }
-
-        $this->removeOptions($product, $attribute, $attributeOptions, $options['locale'], $options['scope']);
+        $this->removeOptions($product, $attribute, $data, $options['locale'], $options['scope']);
     }
 
     /**
      * @param ProductInterface   $product
      * @param AttributeInterface $attribute
-     * @param array              $attributeOptions
+     * @param string[]           $optionCodes
      * @param string|null        $locale
      * @param string|null        $scope
      */
     protected function removeOptions(
         ProductInterface $product,
         AttributeInterface $attribute,
-        $attributeOptions,
+        $optionCodes,
         $locale,
         $scope
     ) {
         $productValue = $product->getValue($attribute->getCode(), $locale, $scope);
 
         if (null !== $productValue) {
-            foreach ($attributeOptions as $attributeOption) {
-                $productValue->removeOption($attributeOption);
+            $product->removeValue($productValue);
```

Comment:
(and maybe, inside the builder, we should rename `addProductValue` to `addOrReplaceProductValue` to be more explicit)